### PR TITLE
Fixed method to check in which cloud environment the cli is running

### PR DESCRIPTION
**Pull request by alvaro-shopify**
### WHY are these changes introduced?
- Fixes the method that checks if the CLI is run inside a cloud environment. Among other things, it is used by the CLI to select the type of authentication it must be used. Right now, in case the CLI is run inside a cloud environment, the device authentication method must be used. The problem now is that the normal authentication flow is being used in cloud environments so it is not working.

### WHAT is this pull request doing?
- Environment variables variables used to verify the cloud environments are not booleans, so checking their existence is should be just enough as a validation

### How to test your changes?

1. Fork the CLI in your own repo
2. Add CLI repo as a remote to checkout the branch `git remote add coworker git://github.com/Shopify/cli.git`
3. Checkout the branch and push it your own repo
   - `git fetch cli bugfix/fix_checking_cloud_environments`
   - `git checkout bugfix/fix_checking_cloud_environments`
4. `git push origin bugfix/fix_checking_cloud_environments`
5. Open this link in the browser (modify it with your repo data) `https://gitpod.io/#https://github.com/<oyu_repo>/<forked_porject>`
6. Checkout the branch ` git checkout bugfix/fix_checking_cloud_environments`
7. Run `yarn shopify app dev --path ./fixtures/app`

NOTE: Skip first five step and go directly to 6th using my repo `https://gitpod.io/#https://github.com/alvaro-shopify/cli`

### Measuring impact

How do we know this change was effective? Please choose one:

- [x] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [ ] I've considered possible [documentation](https://shopify.dev) changes
- [ ] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

File: packages/cli-kit/src/environment/local.test.ts [hunk 1]
```diff
@@ -178,7 +178,7 @@ describe('cloudEnvironment', () => {
     ['GITPOD_WORKSPACE_URL', 'gitpod'],
   ])('returns correct cloud platform when %s is truthy', (envVar, platform) => {
```
Comment (gonzaloriestra) [inline packages/cli-kit/src/environment/local.test.ts:179]:
```suggestion
  ])('returns correct cloud platform when %s exists', (envVar, platform) => {
```

**Review by Arkham — APPROVED** [review]
LGTM, thanks :)

File: packages/cli-kit/src/environment/local.test.ts (hunk 1 above)
Comment (alvaro-shopify) [reply packages/cli-kit/src/environment/local.test.ts:179]: `isTruthy` validates possible values that matches a boolean value. In this case, the values are homogeneous, so checking their existence is enough to consider them as a cloud environment

Comment (isaacroldan) [reply packages/cli-kit/src/environment/local.test.ts:179]: Actually, do you think this works for all systems? `GITPOD_WORKSPACE_URL` will provide a URL, not a truthy value...

Comment (alvaro-shopify) [reply packages/cli-kit/src/environment/local.test.ts]: It is working for the three cloud environments cli are currently supporting:
* gitpod
* codespaces
* spin
It's only checking that the environment variable exists. Maybe we could add a different validation depending on the value type

Comment (isaacroldan) [reply packages/cli-kit/src/environment/local.test.ts]: ah yeah, sorry i misread

File: packages/cli-kit/src/environment/local.ts [hunk 2]
```diff
@@ -127,10 +127,10 @@ export function cloudEnvironment(env: NodeJS.ProcessEnv = process.env): {
   if (isTruthy(env[constants.environmentVariables.codespaces])) {
     return {platform: 'codespaces', editor: true}
   }
-  if (isTruthy(env[constants.environmentVariables.gitpod])) {
+  if (isSet(env[constants.environmentVariables.gitpod])) {
```
Comment (isaacroldan) [inline packages/cli-kit/src/environment/local.ts:130]: I'd use `isSet` for codespaces too

**Pull request merged**